### PR TITLE
Null safety migration of packages/flutter_tools/test/general.shard, part 2/2

### PR DESCRIPTION
**Pull request by liamappelbe**
Null safety migration of packages/flutter_tools/test/general.shard, part 2/2

### CODE REVIEW

**Comment by liamappelbe** [comment]
Blocked on [dwds migration](https://github.com/dart-lang/webdev/issues/1327)

**Comment by liamappelbe** [comment]
Blocked on [dwds migration](https://github.com/dart-lang/webdev/issues/1327)

**Comment by christopherfujino** [comment]
Test failures (commenting so I can keep track myself, please re-ping me when ready for review)

**Comment by liamappelbe** [comment]
Tests are passing

File: packages/flutter_tools/test/general.shard/hot_test.dart [hunk 1]
```diff
@@ -349,7 +348,7 @@ void main() {
           },
         );
 
-        (fakeFlutterDevice.devFS as FakeDevFs).baseUri = Uri.parse('file:///base_uri');
+        (fakeFlutterDevice.devFS as FakeDevFs?)!.baseUri = Uri.parse('file:///base_uri');
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/hot_test.dart:351]:
```suggestion
        (fakeFlutterDevice.devFS! as FakeDevFs).baseUri = Uri.parse('file:///base_uri');
```

File: packages/flutter_tools/test/general.shard/hot_test.dart [hunk 2]
```diff
@@ -279,7 +278,7 @@ void main() {
...
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/hot_test.dart:281]:
```suggestion
        (fakeFlutterDevice.devFS! as FakeDevFs).baseUri = Uri.parse('file:///base_uri');
```

File: packages/flutter_tools/test/general.shard/resident_web_runner_test.dart [hunk 3]
```diff
@@ -1192,7 +1190,9 @@ flutter:
     fakeVmServiceHost = FakeVmServiceHost(requests: <VmServiceExpectation>[]);
     setupMocks();
 
-    webDevFS.exception = ChromeDebugException(<String, dynamic>{});
+    webDevFS.exception = ChromeDebugException(<String, dynamic>{
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/resident_web_runner_test.dart:1193]:
```suggestion
    webDevFS.exception = ChromeDebugException(<String, Object?>{
```

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**